spark/v3.4/spark/src/main/java/org/apache/iceberg/spark/source/SparkBatch.java

```diff
@@ -115,11 +115,11 @@ private String[][] computePreferredLocations() {
   public PartitionReaderFactory createReaderFactory() {
     if (useParquetBatchReads()) {
       int batchSize = readConf.parquetBatchSize();
-      return new SparkColumnarReaderFactory(batchSize);
+      return new SparkColumnarReaderFactory(batchSize, readConf.getReaderType());
```

Comment:
We will need to think whether a new class holding batchSize, reader type, and any future variables makes sense. Let me think about this and come back tomorrow.